cron/polygon/standalone-runner.js

```diff
@@ -166,45 +232,53 @@ async function main() {
-        ethProfit,
-        addresses.pWETH,
-        addresses.V2.quickswap_ETH_MATIC.Underlying,
-      )
-
-      console.log('gasUsed:            ', tx.gasUsed)
-      console.log('profit in Matic:    ', roughProfitInMatic / 1e18)
-      console.log('Matic cost:         ', maticCost / 1e18)
-
-      console.log('[ Is the profit share good enough? ]')
-      console.log('before:             ', ethInProfitShareBefore)
-      console.log('after:              ', ethInProfitShareAfter)
-      console.log('profit shared Ether: ', ethProfit / 1e18)
-
-      if (useDenProfitCalculation(vaultAddress)) {
-        console.log('[ Using DEN profit calculation.. ]')
-        let denInMsigAfter = await den.methods.balanceOf(addresses.msig).call()
-        denProfit = denInMsigAfter - denInMsigBefore
-        console.log('profit shared DEN: ', denProfit / 1e18)
-      }
-
-      if (roughProfitInMatic > maticCost * greatDealRatio || denProfit > minDenProfit) {
-        console.log('====> Time to doHardwork! ====')
-        executeFlag = true
-      } else {
-        console.log('............................. bad deal')
+      try {
+        console.time('doHardwork simulation')
+        let tx = await controller.methods.doHardWork(vaultAddress).send(txSenderInfo)
+        console.timeEnd('doHardwork simulation')
+
+        let maticCost = tx.gasUsed * txSenderInfo.gasPrice
+        let ethInProfitShareAfter = await eth.methods.balanceOf(profitShareAddr).call()
+        ethProfit = ethInProfitShareAfter - ethInProfitShareBefore
+        let roughProfitInMatic = await roughQuoteXInMATIC(
+          ethProfit,
+          addresses.pWETH,
+          addresses.V2.quickswap_ETH_MATIC.Underlying,
+        )
+
+        console.log('gasUsed:            ', tx.gasUsed)
+        console.log('profit in Matic:    ', roughProfitInMatic / 1e18)
+        console.log('Matic cost:         ', maticCost / 1e18)
+
+        console.log('[ Is the profit share good enough? ]')
+        console.log('before:             ', ethInProfitShareBefore)
+        console.log('after:              ', ethInProfitShareAfter)
+        console.log('profit shared Ether: ', ethProfit / 1e18)
+
+        if (useDenProfitCalculation(vaultAddress)) {
+          console.log('[ Using DEN profit calculation.. ]')
+          let denInMsigAfter = await den.methods.balanceOf(addresses.msig).call()
+          denProfit = denInMsigAfter - denInMsigBefore
+          console.log('profit shared DEN: ', denProfit / 1e18)
+        }
+
+        if (roughProfitInMatic > maticCost * greatDealRatio || denProfit > minDenProfit) {
+          console.log('====> Time to doHardwork! ====')
+          executeFlag = true
+        } else {
+          console.log('............................. bad deal')
+        }
+      } catch (e) {
+        console.log('Error during simulation: ')
+        console.log(e)
+        if (settings.prometheusMonitoring && settings.prometheusMonitoring.enabled === true) {
```

Comment:
it can be better to move these two checks into `reportError` so that we do not forget to add it everywhere.